### PR TITLE
feat: support serveTls & serve with ServeTlsOptions

### PR DESCRIPTION
**Pull request by ghost**
https://github.com/denoland/fresh/issues/901
https://github.com/denoland/fresh/issues/809

### CODE REVIEW

**Comment by jiawei397** [comment]
I need to use this function in my project now. When will it be merged?

**Comment by rnbguy** [comment]
hey @lucacasonato any update on this PR? Would love to see this merged :slightly_smiling_face:

File: src/server/mod.ts [hunk 1]
```diff
@@ -60,13 +61,43 @@ export interface DenoConfig {
 
 export { ServerContext };
 
-export async function start(routes: Manifest, opts: StartOptions = {}) {
-  const ctx = await ServerContext.fromManifest(routes, opts);
+export async function start(
+  routes: Manifest,
+  opts: StartOptions | TlsStartOptions = {},
```
Comment (rnbguy) [inline src/server/mod.ts:66]: I would prefer to have two separate `start(..., opts: StartOptions)` and `startTls(..., opts: StartTlsOptions)`.

I would be happy to create a new PR.

Comment (rnbguy) [reply src/server/mod.ts:66]: I cooked up a tls branch on my fork with a straight-forward [patch](https://github.com/denoland/fresh/compare/main...rnbguy:denoland-fresh:tls)

Comment (iuioiua) [reply src/server/mod.ts:66]: I think we should keep the one function, `start()`, that accepts both interfaces. This keeps it in line with `Deno.serve()`, which will soon be made stable.

Comment (rnbguy) [reply src/server/mod.ts:66]: Ah. I missed that `Deno.serveTls()` doesn't exist.

If we are following `Deno.serve()` pattern, and not `serve()`/`serveTls()` pattern from `std/http/server.ts`, it makes sense to keep one `start()`.

Comment (iuioiua) [reply src/server/mod.ts:66]: For context, `serve()` and `serveTls()` from `std` look like they'll be removed once `Deno.serve()` is stabilised (PR [here](https://github.com/denoland/deno_std/pull/3381)).

**Pull request closed without merging**